src/mbgl/map/map.cpp

```diff
@@ -861,6 +870,10 @@ void Map::onLowMemory() {
     impl->view.invalidate();
 }
 
+void Map::Impl::onSourceAttributionChanged(style::Source&, std::string) {
```

Comment:
I struggled with this one. Semantically, I think it makes plenty of sense for the string to go along with the callback, for consistency with the other callbacks. But we aren’t currently plumbing the string all the way through because the `notifyMapChange()` mechanism doesn’t support additional data beyond a single `MapChange` enum.